package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,371 @@
+        decomposition on it.
+    transform(num_eigenvectors)
+        Perform an embedding of a frame into the eigenvectors representing
+        the collective coordinates.
+    """
+
+    def __init__(self, u, epsilon=1, manifold_density=None,
+                 timescale=1, **kwargs):
+        """
+        Parameters
+        -------------
+        u : MDAnalysis Universe or DistanceMatrix object
+            Can be a Universe, in which case one must supply kwargs for the
+            initialization of a DistanceMatrix. Otherwise, this can be a
+            DistanceMatrix already initialized. Either way, this will be made
+            into a diffusion kernel.
+        epsilon : Float
+            Specifies the method used for the choice of scale parameter in the
+            diffusion map. More information in [1], [2] and [3], Default: 1.
+        manifold_density: list, optional
+            The list has to have the same length as the trajectory.
+            With 'None' the weight of each frame of the trajectory will be the
+            same, Default : None
+        timescale: int, optional
+            The number of steps in the random walk, large t reflects global
+            structure whereas small t indicates local structure, Default: 1
+            """
+        if isinstance(u, mda.Universe):
+            self._dist_matrix = DistanceMatrix(u, **kwargs)
+        elif isinstance(u, DistanceMatrix):
+            self._dist_matrix = u
+        else:
+            raise ValueError("U is not a Universe or DistanceMatrix and"
+                             " so the DiffusionMap has no data to work with.")
+        self._epsilon = epsilon
+        # important for transform function and length of .run() method
+        self._nframes = self._dist_matrix.nframes
+        if self._nframes > 2000:
+            warnings.warn("The distance matrix is very large, and can "
+                          "be very slow to compute. Consider picking a larger "
+                          "step size in distance matrix initialization.")
+
+        # determines length of diffusion process
+        self._t = timescale
+
+        if manifold_density is None:
+            # weights do not apply to metric but density of data
+            self._weights_ker = np.ones((self._nframes,))
+        else:
+            if manifold_density.shape[0] != self._nframes:
+                raise ValueError("The weight should have the same length as "
+                                 'the trajectory')
+            else:
+                # density weights are constructed as relative to the mean
+                self._weights_ker = (np.asarray(manifold_density,
+                                     dtype=np.float64) /
+                                     np.mean(manifold_density))
+
+    def run(self):
+        # will only run if distance matrix not already calculated
+        self._dist_matrix.run()
+        self._scaled_matrix = self._dist_matrix.dist_matrix / self._epsilon
```

Comment:
Are you looking at the Lafon paper and seeing h(||x-y||^2/ epsilon)? This is the thing you made me fix a while back. ||x-y||^2 is the metric that they use in the paper, but if I am not mistaken our metric d(x,y) satisfies this kernel property because it is symmetric, positive-definite.